### PR TITLE
Fixed #8065 -- Allowed QuerySet.in_bulk() to accept no arguments.

### PR DESCRIPTION
**Pull request by bxm156**
Modified in_bulk() to return all objects in the queryset when called
with no arguments.

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: docs/ref/models/querysets.txt [hunk 1]
```diff
@@ -1843,6 +1843,16 @@ Example::
 
 If you pass ``in_bulk()`` an empty list, you'll get an empty dictionary.
 
+  .. versionchanged:: 1.10
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: Try to make the annotation "self-contained" as described in https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features.  This line shouldn't be indented either.

File: tests/or_lookups/tests.py [hunk 2]
```diff
@@ -234,3 +234,11 @@ def test_other_arg_queries(self):
             Article.objects.filter(Q(headline__startswith='Hello')).in_bulk([self.a1, self.a2]),
             {self.a1: Article.objects.get(pk=self.a1)}
         )
+
+        self.assertEqual(
```
Comment (timgraham) [inline tests/or_lookups/tests.py]: Does this test cover any changed code that the other test doesn't cover? If not, I'd omit it.

File: docs/releases/1.10.txt [hunk 3]
```diff
@@ -175,6 +175,9 @@ Models
   accessible as a descriptor on the proxied model class and may be referenced in
   queryset filtering.
 
+* :meth:`QuerySet.in_bulk() <django.db.models.query.QuerySet.in_bulk>`
+  will return all objects in a queryset if no arguments are given.
```
Comment (timgraham) [inline docs/releases/1.10.txt]: "may be called without any arguments to return all objects in the queryset."

File: docs/ref/models/querysets.txt [hunk 4]
```diff
@@ -1843,6 +1843,16 @@ Example::
...
+
+    If you do not pass in any arguments, ``in_bulk()`` will return all
+    objects in the queryset.
+
+    Example::
+
+        >>> Blog.objects.in_bulk()
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: You can put this with the other examples I think.

File: django/db/models/query.py [hunk 5]
```diff
@@ -561,16 +561,19 @@ def last(self):
             return objects[0]
         return None
 
-    def in_bulk(self, id_list):
+    def in_bulk(self, id_list=None):
         """
         Returns a dictionary mapping each of the given IDs to the object with
-        that ID.
+        that ID. If called with no arguments, the entire queryset is evaluated.
```
Comment (timgraham) [inline django/db/models/query.py]: without any arguments

**Comment by timgraham** [comment]
Please squash commits and uncheck "Patch needs improvement" on the ticket when you update the pull request, thanks!

**Comment by bxm156** [comment]
Thanks for the feedback! I will get to it later this week

**Comment by timgraham** [comment]
Hi Bryan, any plan to return to this soon or should we close it for now?

**Comment by bxm156** [comment]
Hi,  sorry about the delay. I'll finish it this weekend.

**Comment by timgraham** [comment]
merged in 62ca2dea04489bdb23df1c7815439287c6d44630, thanks!

**Pull request closed without merging**